server.go

```diff
@@ -85,15 +85,36 @@ func Listen(conn net.PacketConn, config *Config) (Listener, error) {
 	return s, nil
 }
 
+var defaultSourceAddressVerification = func(clientAddr net.Addr, stk *STK) bool {
+	if stk == nil {
+		return true
+	}
+	if time.Now().After(stk.sentTime.Add(protocol.STKExpiryTime)) {
+		return true
+	}
+	var sourceAddr string
+	if udpAddr, ok := clientAddr.(*net.UDPAddr); ok {
+		sourceAddr = udpAddr.IP.String()
+	} else {
+		sourceAddr = clientAddr.String()
+	}
+	return sourceAddr != stk.remoteAddr
```

Comment:
I didn't have anything specific in mind, but it's a good idea to always use constant time comparisons for secret values, as there's practically no disadvantage.